src/Intent.cpp

```diff
@@ -25,17 +25,48 @@
 
 using namespace jni;
 
-std::string CJNIIntent::EXTRA_KEY_EVENT;
+std::string CJNIIntent::ACTION_BATTERY_CHANGED;
+std::string CJNIIntent::ACTION_BOOT_COMPLETED;
+std::string CJNIIntent::ACTION_DREAMING_STOPPED;
+std::string CJNIIntent::ACTION_GET_CONTENT;
+std::string CJNIIntent::ACTION_HEADSET_PLUG;
+std::string CJNIIntent::ACTION_MAIN;
+std::string CJNIIntent::ACTION_MEDIA_BUTTON;
+std::string CJNIIntent::ACTION_MEDIA_MOUNTED;
+std::string CJNIIntent::ACTION_MEDIA_REMOVED;
+std::string CJNIIntent::ACTION_MEDIA_SHARED;
+std::string CJNIIntent::ACTION_MEDIA_UNMOUNTABLE;
+std::string CJNIIntent::ACTION_MEDIA_UNMOUNTED;
 std::string CJNIIntent::ACTION_OPEN_DOCUMENT_TREE;
+std::string CJNIIntent::ACTION_SCREEN_ON;
+std::string CJNIIntent::ACTION_SCREEN_OFF;
+std::string CJNIIntent::ACTION_SEARCH;
+std::string CJNIIntent::ACTION_VIEW;
+std::string CJNIIntent::EXTRA_KEY_EVENT;
 
 static std::string s_className = "android/content/Intent";
 
 void CJNIIntent::PopulateStaticFields()
 {
   jhclass clazz = find_class(s_className.c_str());
+  ACTION_BATTERY_CHANGED = jcast<std::string>(get_static_field<jhstring>(clazz,"ACTION_BATTERY_CHANGED"));
```

Comment:
Maybe use a macro to avoid c/p errors?

Reply:
When mapping class variables we use the same names, this way it's easy to check that they are correct and therefore the probability of error is reduced, for the moment I don't think it's necessary